node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/nodeadmin/NodeAdminStateUpdater.java

```diff
@@ -257,28 +286,22 @@ private void fetchContainersToRunFromNodeRepository() {
     }
 
     public void start() {
-        if (loopThread != null) {
-            throw new RuntimeException("Can not restart NodeAdminStateUpdater");
-        }
-
-        loopThread = new Thread(() -> {
-            while (! terminated.get()) tick();
-        });
-        loopThread.setName("tick-NodeAdminStateUpdater");
         loopThread.start();
-
-        specVerifierScheduler.scheduleWithFixedDelay(() ->
-                updateHardwareDivergence(storageMaintainer), 5, 60, TimeUnit.MINUTES);
     }
 
     public void stop() {
-        specVerifierScheduler.shutdown();
+        log.info(objectToString() + ": Stop called");
         if (!terminated.compareAndSet(false, true)) {
             throw new RuntimeException("Can not re-stop a node agent.");
         }
 
+        synchronized (classLocking) {
+            classLocking.notifyAll();
```

Comment:
We should not call the builtin Object::notifyAll to avoid exposing implementation details.

Reply:
Fixed.